### PR TITLE
fix warnings in all codebase

### PR DESCRIPTION
**Pull request by alebaffa**

### CODE REVIEW

File: pallets/accounts/src/tests.rs [hunk 1]
```diff
@@ -215,118 +214,9 @@ mod unlink_tests {
 mod sync_partner_contracts_tests {
 	use super::*;
 
-	fn hardcode_expected_request_and_response(
-		state: &mut testing::OffchainState,
-		lock: Lock,
-	) -> u64 {
-		let geth_url = Some(String::from("https://www.dummywebsite.com/"));
```
Comment (sinkingsugar) [inline pallets/accounts/src/tests.rs]: wow why? are we sure we dont use this?

**Review by sinkingsugar — CHANGES_REQUESTED** [review]

File: pallets/accounts/src/tests.rs (hunk 1 above)
Comment (alebaffa) [reply pallets/accounts/src/tests.rs]: I put it back. It is used in a test that needs to be fixed, so ignored for the moment (and so the call to this is temporarily removed) and that is why cargo saw it as unused. I forgot about this. I will fix the error in this PR too.

**Comment by sinkingsugar** [comment]
Since you used a merge commit please squash when merging 👍

File: pallets/accounts/src/tests.rs (hunk 1 above)
Comment (alebaffa) [reply pallets/accounts/src/tests.rs]: Fixed.

**Review by sinkingsugar — APPROVED** [review]

**Pull request merged**